tests/functional/workloads/cnv/test_vm_snapshot_cloning_ops.py

```diff
@@ -290,3 +291,131 @@ def test_vm_snapshot_ops(
+        except Exception as e:
+            log.error(f"[{vm_obj.name}] Error during VM processing: {e}", exc_info=True)
+            raise
+
+    def run_parallel_vm_clone_restore(
+        self,
+        vm_list,
+        file_paths,
+        admin_client,
+        vm_clone_fixture,
+        vm_snapshot_restore_fixture,
+    ):
+        """
+        Process operations on VMs in parallel including cloning, snapshot restore of cloned VM, and data checksum.
+
+        Args:
+            vm_obj (object): The virtual machine object to operate on.
+            file_paths (list): List of file paths to be handled or verified during the operation.
+            admin_client (object): The admin client instance.
+            vm_clone_fixture (fixture): Pytest fixture used to clone the VM.
+            vm_snapshot_restore_fixture (fixture): Pytest fixture used to create and restore VM snapshots.
+
+        """
+
+        MAX_WORKERS = 10
+        with ThreadPoolExecutor(max_workers=MAX_WORKERS) as executor:
+            futures = {
+                executor.submit(
+                    config_safe_thread_pool_task,
+                    config.default_cluster_index,
+                    self.process_vm_with_snapshot_clone,
+                    vm_obj,
+                    file_paths,
+                    admin_client,
+                    vm_clone_fixture,
+                    vm_snapshot_restore_fixture,
+                ): vm_obj.name
+                for vm_obj in vm_list
+            }
+            for future in as_completed(futures):
+                vm_name = futures[future]
+                try:
+                    future.result()
+                except Exception as e:
+                    log.error(
+                        f"[{vm_name}] Exception occurred during test execution: {e}"
+                    )
+                    raise
+
+    @workloads
+    @pytest.mark.polarion_id("OCS-6325")
+    def test_vm_snap_of_clone(
+        self,
+        setup_cnv,
+        project_factory,
+        multi_cnv_workload,
+        vm_clone_fixture,
+        vm_snapshot_restore_fixture,
+        admin_client,
+    ):
+        """
+        This test performs the VM cloning and IOs created using different volume interfaces(PVC/DV/DVT)
+
+        Test steps:
+        1. Create a clone of a VM by following the documented procedure from CNV official docs.
+        2. Add additional data to the cloned VM.
+        3. Create snapshot of cloned VM
+        4. Vertify snapshot of cloned VM created successfully.
+        5. Check data conisistency on the Restored VM
+        6. Repeat the above procedure for all the VMs in the system
+        7. Delete all the clones and restored VM created as part of this test
+        """
+
+        proj_obj = project_factory()
+        file_paths = ["/source_file.txt", "/new_file.txt"]
+        vm_objs_def, vm_objs_aggr, _, _ = multi_cnv_workload(
+            namespace=proj_obj.namespace
+        )
+
```

Comment:
there are many empty lines in this pr. I feel there is no need for those, let me know if you think otherwise